### ISSUE TITLE
Support state sync

### ISSUE THREAD

**Issue by ethanfrey** [CosmWasm/wasmd#478]
State sync will copy over the iavl store, but it has no idea of the contracts stored under `data/wasm`. This leads to the following error when querying a contract on a node that was state synced:

```
Error: Error calling the VM: Cache error: Error opening Wasm file for reading: No such file or directory (os error 2): query wasm contract failed: invalid request
```

Requires a solution to https://github.com/cosmos/cosmos-sdk/issues/7340 before we can address this.

**Comment by ethanfrey** [comment]
Update - there seem to be a push from other chains as well and this is slated for a v0.43.x patch release

**Comment by alpe** [comment]
fyi: Secret hat an [implementation ](https://github.com/scrtlabs/SecretNetwork/blob/v1.3.0-beta.10/x/compute/wasm_snapshotter.go)

**Comment by ethanfrey** [comment]
Closed by #823
